<a href="https://colab.research.google.com/github/Python6868/fileopen/blob/main/baitapdanalysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Bài Tập Lab2


#Bài 1

In [6]:
# Khai báo biến int
int_1 = 10
int_2 = 100
int_3 = -789
# Khai báo biến fload
float_1 = 0.0
float_2 = 15.20
float_3 = -.9
# Khai báo biến complex
complex_1 = 3.14j
complex_2 = 45.j
complex_3 = 9.322e-36j
print(f"{'int':^10}{'float':^10}{'complex'}")
for i in range(1, 4):
  print(f"{locals()[f'int_{i}']:<10}{locals()[f'float_{i}']:<10}{locals()[f'complex_{i}']:<10}")


   int      float   complex
10        0.0       3.14j     
100       15.2      45j       
-789      -0.9      9.322e-36j


# Bài 2
Chuyển các giá trị sau ra số nguyên (int)
12.365
“15.99”
-13.5

Chuyển các xâu sau ra số thực
“-32.54e6”
“-.9e6”


In [16]:
# Chuyển các giá trị sau ra số nguyên (int)
x = 12.365
y = "15.99"
z = -13.5
print(int(x))
print(int(float(y)))
print(int(z))
# Chuyển các xâu ra số thực
xau_1 = "-32.54e6"
xau_2 = "-.9e6"
print(float(xau_1))
print(float(xau_2))

12
15
-13
-32540000.0
-900000.0


# Bài 3
Sử dụng hàm math.ceil và math.floor trên 4 giá trị 15.11; 15.99; -15.11; -15.99. Từ đó bạn cần rút ra nhận xét về 2 hàm này

Tính round(5.4), round(5.5), round(5.6) và round(-5.4)

Làm tròn đến 2 chữ số cho các giá trị 5.431; 5.435; và 5.439. Xem kết quả và bạn cần rút ra nhận xét về hàm round.


In [10]:
import math
# Sử dụng hàm math.ceil
print(math.ceil(15.11))
print(math.ceil(15.99))
print(math.ceil(-15.11))
print(math.ceil(-15.99))
print()
# Sử dụng hàm math.floor
print(math.floor(15.11))
print(math.floor(15.99))
print(math.floor(-15.11))
print(math.floor(-15.99))
# Nhận xét hàm math.ceil làm tròn lên với số dương, số âm ngược lại
print(round(5.4))
print(round(5.5))
print(round(5.6))
print()
print(round(-5.4))
print(round(-5.5))
print(round(-5.6))
print()
print(round(5.431,2))
print(round(5.435,2))
print(round(5.439,2))

16
16
-15
-15

15
15
-16
-16
5
6
6

-5
-6
-6

5.43
5.43
5.44


🔥 Nhận xét
math.ceil(x): Làm tròn lên đến số nguyên lớn hơn hoặc bằng x.
🔹 Dương: math.ceil(15.11) → 16, math.ceil(15.99) → 16
🔹 Âm: math.ceil(-15.11) → -15, math.ceil(-15.99) → -15

math.floor(x): Làm tròn xuống đến số nguyên nhỏ hơn hoặc bằng x.
🔹 Dương: math.floor(15.11) → 15, math.floor(15.99) → 15
🔹 Âm: math.floor(-15.11) → -16, math.floor(-15.99) → -16

📌 Kết luận:

Dương: ceil() luôn làm tròn lên, floor() làm tròn xuống.
Âm: ceil() làm tròn lên (đến số nhỏ hơn về giá trị tuyệt đối), floor() làm tròn xuống (đến số nhỏ hơn hơn thực tế).
**********************
🔥 Nhận xét về round()
round(x) làm tròn đến số nguyên gần nhất.
Nếu .5, Python sẽ làm tròn về số chẵn gần nhất!
round(5.5) → 6 (vì 6 là số chẵn gần nhất)
round(4.5) → 4 (vì 4 là số chẵn gần nhất)
📌 Kết luận: round() sử dụng quy tắc làm tròn đến số chẵn gần nhất khi .5 (Bankers' rounding)
*********************
🔥 Nhận xét
round(5.431, 2) → 5.43 (làm tròn xuống)
round(5.435, 2) → 5.44 (làm tròn lên do .5)
round(5.439, 2) → 5.44 (làm tròn lên)
📌 Kết luận:

round(x, n) làm tròn đến n chữ số thập phân.
Khi .5, Python làm tròn lên.
************************
🎯 Tóm tắt quan trọng
Hàm	Chức năng	Ví dụ
math.ceil(x)	Làm tròn lên	math.ceil(3.1) → 4
math.floor(x)	Làm tròn xuống	math.floor(3.9) → 3;
round(x)	Làm tròn về số nguyên gần nhất	round(5.5) → 6;
round(x, n)	Làm tròn đến n chữ số thập phân	round(5.435, 2) → 5.44

In [ ]:
print(12345)
print(6)

# Bài 4
In ra kết quả sau: modf(4.56)
Kết quả trả về là kiểu dữ liệu gì?
Từ kết quả trả về, lưu phần nguyên và phần thực ra 2 biến riêng biệt, và in ra giá trị của 2 biến này


In [18]:
import math
print(math.modf(4.56))
for x in math.modf(4.56):
  print(x)

(0.5599999999999996, 4.0)
0.5599999999999996
4.0


# Bài 5

Tính max, min của bộ 3 giá trị: 2^(3^4 ), 3^(2^4 ), và 4^(3^2 )




In [20]:
value = [2**(3**4), 3**(2**4), 4**(3**2)]
print("Giá trị max của bộ 3 giá trị: ", max(value))
print("Giá trị min của bộ 3 giá trị: ", min(value))


Giá trị max của bộ 3 giá trị:  2417851639229258349412352
Giá trị min của bộ 3 giá trị:  262144


# Bài 6
Cho xâu s = “The best of the BBC, with the latest news and sport headlines, weather, TV & radio highlights and much more from across”

Cần xử lý sao cho tách được xâu con x, từ chữ “with” đến hết xâu s. In xâu x ra màn hình

Lưu ý: x = “with the latest news and sport headlines, weather, TV & radio highlights and much more from across”


In [26]:
s = "The best of the BBC, with the latest news and sport headlines, weather, TV & radio highlights and much more from across"
print(s[21:])

with the latest news and sport headlines, weather, TV & radio highlights and much more from across


# Bài 7
In dòng chữ sau ra màn hình

Hello everyone. My barcode's \44GFD\


In [29]:
print("Hello everyone. My barcode's \\44GFD\\")

Hello everyone. My barcode's \44GFD\


# Bài 8
Cho một list:

list = [“Nguyễn Văn Nam”, “Trần Xuân Hải”, “Lương Thế Anh”]

Bạn cần làm các nhiệm vụ sau:

•	Xóa 2 phần tử đầu tiên của list

•	In list sau khi xóa

•	Thêm lại 2 phần tử đã xóa, sao cho list mới có thứ tự tên xếp theo A-B-C (Lương Thế Anh là phần tử đầu tiên, sau đó đến Trần Xuân Hải và sau cùng là Nguyễn Văn Nam)

•	In lại list sau khi thêm 2 phần tử



In [44]:
list = ["Nguyễn Văn Nam", "Trần Xuân Hải", "Lương Thế Anh"]
print(list)
del list[:2]
print(list)
list.insert(1, "Trần Xuân Hải")
list.insert(2, "Nguyễn Văn Nam")
print(list)

['Nguyễn Văn Nam', 'Trần Xuân Hải', 'Lương Thế Anh']
['Lương Thế Anh']
['Lương Thế Anh', 'Trần Xuân Hải', 'Nguyễn Văn Nam']


# Bài 9
Tạo 1 tập hợp (set) s1 với các phần tử là “Nam”, “Hải”, “Hoàng”, “Hùng”, “An”

In tập hợp s1 ra màn hình

Thêm vào tập hợp s1 phần tử “Bắc”. In lại s1

Xóa phần tử “An” ra khỏi tập hợp. Bạn sẽ dùng phương thức discard hay remove, tại sao? In lại s1?

Sau cùng, thêm vào tập hợp s1 đồng thời 4 phần tử là “Hải”, “Tuấn”, “Dũng”, “Bắc”. Theo bạn sau cùng, s1 sẽ có bao nhiêu phần tử, là những phần tử nào. In kết quả s1 ra màn hình để kiểm tra.


In [59]:
s1 = {"Nam", "Hải", "Hoàng", "Hùng", "An"}
print(s1)
s1.add("Bắc")
print(s1)
s1.remove("An")
print(s1)
s1.update(["Hải", "Tuấn", "Dũng", "Bắc"])
print(s1)
n = len(s1)
print(n)
for x in s1:
  print(x)


{'Hoàng', 'Hùng', 'Nam', 'Hải', 'An'}
{'Hoàng', 'Hùng', 'Nam', 'Hải', 'An', 'Bắc'}
{'Hoàng', 'Hùng', 'Nam', 'Hải', 'Bắc'}
{'Hoàng', 'Hùng', 'Dũng', 'Nam', 'Hải', 'Tuấn', 'Bắc'}
7
Hoàng
Hùng
Dũng
Nam
Hải
Tuấn
Bắc


# Bài 10
Sử dụng kiểu dữ liệu từ điển, tạo một cuốn danh bạ điện thoại có cấu trúc như sau. In danh bạ ban đầu ra màn hình.

{
“Nguyễn Văn Hùng”: {“SĐT”: “0913212454”, “email”: “hungnv@gmail.com”, “address”: “23 Nguyễn Trãi, Thanh Xuân, Hà Nội”},
“Nguyễn Văn Nam”: {“SĐT”: “0913213454”, “email”: “namnv@gmail.com”, “address”: “15 Đại Cồ Việt, Hai Bà Trưng, Hà Nội”},
}

Xóa “Nguyễn Văn Hùng” ra khỏi danh bạ. In lại danh bạ.


Thêm mới 2 liên hệ khác vào danh bạ điện thoại. In ra danh bạ.


In ra màn hình SĐT của một người nào đó (Ví dụ SĐT của Nguyễn Văn Nam)



In [71]:
data = { "Nguyễn Văn Hùng": {"SĐT": "0913212454", "email": "hungnv@gmail.com", "address": "23 Nguyễn Trãi, Thanh Xuân, Hà Nội"}, "Nguyễn Văn Nam": {"SĐT": "0913213454", "email": "namnv@gmail.com", "address": "15 Đại Cồ Việt, Hai Bà Trưng, Hà Nội"}, }
data = { "Nguyễn Văn Hùng": {"SĐT": "0913212454", "email": "hungnv@gmail.com", "address": "23 Nguyễn Trãi, Thanh Xuân, Hà Nội"}, "Nguyễn Văn Nam": {"SĐT": "0913213454", "email": "namnv@gmail.com", "address": "15 Đại Cồ Việt, Hai Bà Trưng, Hà Nội"}, }
del data["Nguyễn Văn Hùng"]
print(data)
data.update({"Nguyễn Tuấn Anh": {"SĐT": "0913213454", "email": "tuananh@gmail.com", "address": "15 Đại Cồ Việt, Hai Bà Trưng, Hà Nội"}})
print(data)
print("SĐT của Nguyễn Văn Nam: ", data["Nguyễn Văn Nam"]["SĐT"])

{'Nguyễn Văn Nam': {'SĐT': '0913213454', 'email': 'namnv@gmail.com', 'address': '15 Đại Cồ Việt, Hai Bà Trưng, Hà Nội'}}
{'Nguyễn Văn Nam': {'SĐT': '0913213454', 'email': 'namnv@gmail.com', 'address': '15 Đại Cồ Việt, Hai Bà Trưng, Hà Nội'}, 'Nguyễn Tuấn Anh': {'SĐT': '0913213454', 'email': 'tuananh@gmail.com', 'address': '15 Đại Cồ Việt, Hai Bà Trưng, Hà Nội'}}
SĐT của Nguyễn Văn Nam:  0913213454


# Bài 11
In ra kết quả logic của mệnh đề: 2^(3^4 ) lớn hơn 2^(4^3 )

In [72]:
print(bool(2**(3**4) > 2**(4**3)))

True
